### PR TITLE
separate categories.yml from metadata.yml

### LINKED ISSUE #3499: [Icons] reintroduce `categories.yml` for easier editability while iterating
info in `categories.yml` was appended to `metadata.yml` for a simpler build process, but it's a bit harder to author/edit icons with this workflow.

should switch back to two separate files for the meantime while we sort out our contribution workflow

### PR DESCRIPTION
**Pull request by lovemecomputer**
Closes #3499 

**changed**
- removes categories info from `metadata.yml` and puts it in `categories.yml`
- updates `build-metadata.js` to use this new file

**testing**
- run `node tasks/build-metadata.js` 
- verify that `the output file `icons/metadata.json` contains this info at the bottom as expected (should be unchanged)

### CODE REVIEW

File: packages/icons/tasks/build-metadata.js [hunk 1]
```diff
@@ -15,13 +15,21 @@ const yaml = require('js-yaml');
...
+  const categories = yaml.safeLoad(
+    await fs.readFile(CATEGORIES_DEFINITION_PATH, 'utf8')
+  );
 
-  await fs.writeJson(METADATA_OUTPUT, metadata, {
-    spaces: 2,
-  });
+  await fs.writeJson(
+    METADATA_OUTPUT,
+    { ...metadata, ...categories },
```
Comment (joshblack) [inline packages/icons/tasks/build-metadata.js:28]:
```suggestion
    { ...metadata, categories },
```

**Review by joshblack — CHANGES_REQUESTED** [review]

File: packages/icons/tasks/build-metadata.js (hunk 1 above)
Comment (lovemecomputer) [reply packages/icons/tasks/build-metadata.js:28]: @joshblack would this cause the data to nest differently?

Comment (lovemecomputer) [reply packages/icons/tasks/build-metadata.js:28]: @joshblack yeah that seems to add to an extra level of duplicate nesting:
Screen Shot 2019-07-22 at 5 41 54 PM

Comment (joshblack) [reply packages/icons/tasks/build-metadata.js:28]: Got it! Sorry, assumed `categories` was a different structure, looks great 👍

**Review by joshblack — APPROVED** [review]

**Review by emyarod — APPROVED** [review]
looks good to me!

**Pull request merged**